src/AutoMapper.php

```diff
@@ -53,28 +59,21 @@ class AutoMapper implements AutoMapperInterface, AutoMapperRegistryInterface, Ma
     public const RELEASE_VERSION = 0;
     public const EXTRA_VERSION = 'DEV';
 
-    /** @var MapperGeneratorMetadataInterface[] */
-    private array $metadata = [];
-
     /** @var GeneratedMapper[] */
     private array $mapperRegistry = [];
 
     public function __construct(
         private readonly ClassLoaderInterface $classLoader,
         private readonly ChainTransformerFactory $chainTransformerFactory,
-        public readonly CustomTransformersRegistry $customTransformersRegistry,
-        private readonly ?MapperGeneratorMetadataFactoryInterface $mapperConfigurationFactory = null
+        private readonly CustomTransformersRegistry $customTransformersRegistry,
+        private readonly MapperGeneratorMetadataRegistryInterface $mapperGeneratorMetadataRegistry,
+        private readonly MapperGenerator $mapperGenerator,
```

Comment:
I'm just realizing this change might not be clever: this forces to instantiate `MapperGenerator` everytime the automapper is used, but as soon as all mappers are generated, nothing regarding "code generation" should be instantiated...

I've done it this way because, we now have a lot more classes to instantiate, and some tests are very cumbersome to update sometimes now...

I think I'll re-work this whole thing. But for now, let's merge it like this (the target branch is a temporary one).